### ISSUE TITLE
Error on st_makeenvelope signature when generating tiles 

### ISSUE THREAD

**Issue by fnicollet** [Geodan/i3dm.export#14]
Hello and thanks for making this toolset,

I have been trying to use it on simple data (about 31K trees) by following the instructions, and when I get to the tile generation on a Postgresql 12 table:
```
i3dm.export -c "Host=127.0.0.1;Username=postgres;password=postgres;Database=bdx;Port=5432" -t public.arbres_instances -o D:\GeoWeb\bdx-3d-data\i3dm\arbres\
```
I get the following error:
```
Exporting i3dm's from public.arbres_instances...
Tile extent: 1000
0,00% Exporting i3dm tiles...                                           00:00:02
C:\Users\support>

t_makeenvelope(integer, integer, integer, integer, integer, integer, integer, in
teger, integer) n'existe pas
   at Npgsql.NpgsqlConnector.<>c__DisplayClass160_0.<<DoReadMessage>g__ReadMessa
geLong|0>d.MoveNext()
--- End of stack trace from previous location ---
   at Npgsql.NpgsqlConnector.<>c__DisplayClass160_0.<<DoReadMessage>g__ReadMessa
geLong|0>d.MoveNext()
--- End of stack trace from previous location ---
   at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming)
   at Npgsql.NpgsqlDataReader.NextResult()
   at Npgsql.NpgsqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean
async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.System.Data.IDbCommand.ExecuteReader(CommandB
ehavior behavior)
   at Dapper.SqlMapper.ExecuteReaderWithFlagsFallback(IDbCommand cmd, Boolean wa
sClosed, CommandBehavior behavior) in /_/Dapper/SqlMapper.cs:line 1051
   at Dapper.SqlMapper.QueryImpl[T](IDbConnection cnn, CommandDefinition command
, Type effectiveType)+MoveNext() in /_/Dapper/SqlMapper.cs:line 1079
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Dapper.SqlMapper.Query[T](IDbConnection cnn, String sql, Object param, IDb
Transaction transaction, Boolean buffered, Nullable`1 commandTimeout, Nullable`1
 commandType) in /_/Dapper/SqlMapper.cs:line 721
   at i3dm.export.InstancesRepository.GetInstances(NpgsqlConnection conn, String
 geometry_table, Point from, Point to, String query, Boolean useScaleNonUniform)
 in D:\dev\github.com\geodan\i3dm.export\src\InstancesRepository.cs:line 16
   at i3dm.export.Program.CreateTile(Options o, String tileFolder, NpgsqlConnect
ion conn, BoundingBox3D rootBounds, List`1 tiles, Int32 x, Int32 y, String prefi
x) in D:\dev\github.com\geodan\i3dm.export\src\Program.cs:line 96
   at i3dm.export.Program.<>c.<Main>b__0_0(Options o) in D:\dev\github.com\geoda
n\i3dm.export\src\Program.cs:line 68
   at CommandLine.ParserResultExtensions.WithParsed[T](ParserResult`1 result, Ac
tion`1 action)
   at i3dm.export.Program.Main(String[] args) in D:\dev\github.com\geodan\i3dm.e
xport\src\Program.cs:line 18
  Exception data:
    Severity: ERREUR
    SqlState: 42883
    MessageText: la fonction st_makeenvelope(integer, integer, integer, integer,
 integer, integer, integer, integer, integer) n'existe pas
    Hint: Aucune fonction ne correspond au nom donné et aux types d'arguments.
Vous devez ajouter des conversions explicites de type.
    Position: 158
    File: d:\pginstaller_12.auto\postgres.windows-x64\src\backend\parser\parse_f
unc.c
    Line: 631
    Routine: ParseFuncOrColumn
```
The DB message is in french, but it says that the signature `st_makeenvelope(integer, integer, integer, integer, integer, integer, integer, integer, integer)` does not exist. And indeed, the ST_MakeEnvelope signature only takes 5 parameters and not 9:
https://postgis.net/docs/ST_MakeEnvelope.html

When looking at the stack trace, it seems like the function is called here:
https://github.com/Geodan/i3dm.export/blob/main/src/InstancesRepository.cs#L15

It is as if the bound parameter `{from.X}` would be seen as 2 parameters (so 2x4 + 1 would be 9 instead of 5)? I don't know much about .net, so I don't know what could produce such behaviour

Thanks for your help,
Fabien

**Comment by fnicollet** [comment]
If it helps, when I execute the following query:
```
psql -U postgres -d bdx -c "SELECT st_xmin(box), ST_Ymin(box), ST_Zmin(box), ST_Xmax(box), ST_Ymax(box), ST_Zmax(box) FROM (select ST_3DExtent(st_transform(geom, 3857)) AS box from public.arbres_instances) as total";
```
The result is:
```
       st_xmin       |      st_ymin       | st_zmin |      st_xmax      |      st_ymax      |    st_zmax
---------------------+--------------------+---------+-------------------+-------------------+----------------
 -132934.70832944845 | 5565528.2976210695 |       0 | -59824.1173689642 | 5606955.071889523 | 1.2612733e-316
(1 ligne)
```
The dataset is located around Bordeaux, France

Fabien

**Comment by fnicollet** [comment]
I think i have found a workaround. My hunch is as it seems to look for a signature with 9 integers, the decimal numbers are not interpreted correctly. So "3.4" is interpreted as "3", "4". 
So I created a "compatibility function":
```sql
DROP FUNCTION IF EXISTS public.st_makeenvelope(integer,integer,integer,integer,integer,integer,integer,integer,integer);
CREATE OR REPLACE FUNCTION public.st_makeenvelope(int1 integer,int2 integer,int3 integer,int4 integer,int5 integer,int6 integer,int7 integer,int8 integer,srid integer DEFAULT 0)
  RETURNS geometry AS $$
  BEGIN
  --  temp compat function for i3dm.export
	  RETURN ST_MakeEnvelope(
							(int1::text || '.' || int2::text)::double precision,
							(int3::text || '.' || int4::text)::double precision,
							(int5::text || '.' || int6::text)::double precision,
							(int7::text || '.' || int8::text)::double precision,
							srid);
  END;
  $$ language plpgsql COST 1;
```
The process seems to crash a little further because i didn't give the program the right path to the GLB file, so it seems to do the trick. 
I don't why the decimal are not interpreted correctly, maybe because the OS is in French and "," is the french decimal separator?

**Comment by bertt** [comment]
ok could be some decimal separator issue. Another workaround is changing decimal symbol from ',' to '.'  in windows region and language settings.

**Comment by bertt** [comment]
Should be fixed in 825db3e using FormattableString.Invariant
